src/endpoints.ts

```diff
@@ -1,24 +1,44 @@
-import {API, APIClient} from '@wharfkit/antelope'
+import {ABI, APIClient} from '@wharfkit/antelope'
+import {GetVotersResponse} from './types'
 
-export class ExampleAPI {
+export class HyperionAPIClient {
     constructor(private client: APIClient) {}
 
-    /**
-     * Define the calls for the API
-     */
-    // async get_raw_abi(accountName: NameType) {
-    //     return this.call({
-    //         path: '/v1/chain/get_raw_abi',
-    //         params: {account_name: Name.from(accountName)},
-    //         responseType: GetRawAbiResponse,
-    //     })
-    // }
-
-    // Example for testing
-    async get_info() {
+    async get_abi_snapshot(contract: string, block?: number, fetch = false): Promise<ABI.Def> {
+        if (!block) {
+            const info = await this.client.v1.chain.get_info()
+
+            block = Number(info.last_irreversible_block_num)
+        }
+
+        return this.client.call<ABI.Def>({
+            path: `/v2/history/get_abi_snapshot?contract=${encodeURIComponent(
+                contract
+            )}&block=${block}&fetch=${fetch}`,
+            method: 'GET',
+        })
+    }
+
+    async get_voters(
+        producer?: string,
```

Comment:
Should we make all the inputs be typed, e.g. this one being a `NameType`?

Reply:
Yep!